packages/schemas/src/seeds/tenant.ts

```diff
@@ -1,12 +1,21 @@
 import type { InferModelType } from '@withtyped/server';
+import { z } from 'zod';
 
 import type { Tenants } from '../models/tenants.js';
+import { TenantTag } from '../types/tenant.js';
 
 export const defaultTenantId = 'default';
 export const adminTenantId = 'admin';
 
-/**
- * `createModel` from @withtyped/server can not properly infer the model
- * type, manually define it here for now.
- */
-export type TenantModel = Pick<InferModelType<typeof Tenants>, 'id' | 'dbUser' | 'dbUserPassword'>;
+export type TenantModel = InferModelType<typeof Tenants>;
+
+export type PatchTenant = Partial<Pick<TenantModel, 'name' | 'tag'>>;
+export type CreateTenant = Pick<TenantModel, 'id' | 'dbUser' | 'dbUserPassword'> &
+  PatchTenant & { createdAt?: number };
+export const createTenantGuard = z.object({
+  id: z.string(),
+  dbUser: z.string(),
+  dbUserPassword: z.string(),
+  name: z.string().optional(),
+  tag: z.nativeEnum(TenantTag).optional(),
+});
```

Comment:
Should we place all these types and guard definitions under the seeds directory?